h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -2067,15 +2198,27 @@ private void coldStart(double[] devHistoryTrain, double[] devHistoryTest) {
     
     protected Submodel computeSubmodel(int i,double lambda, double nullDevTrain, double nullDevValid) {
       Submodel sm;
-      if(lambda >= _lmax && _state.l1pen() > 0)
-        _model.addSubmodel(sm = new Submodel(lambda, _state.alpha(), getNullBeta(),_state._iter,nullDevTrain, nullDevValid));
-      else {  // this is also the path for HGLM model
-        sm = new Submodel(lambda, _state.alpha(), _state.beta(), _state._iter, -1, -1);// restart from last run
-         if (_parms._HGLM) // add random coefficients for random effects/columns
-          sm.ubeta = Arrays.copyOf(_state.ubeta(), _state.ubeta().length);
-        _model.addSubmodel(sm);
-        if (!_parms._HGLM) // only perform this when HGLM is not used.
-          _state.setLambda(lambda);
+      if(lambda >= _lmax && _state.l1pen() > 0) {
+        if (_parms.hasCheckpoint() && (_parms._alpha.length > 1 || _parms._lambda.length > 1) && 
```

Comment:
- this condition is the same for both branches, it would be more readable if it would be extracted to a local variable

- comparison order should be reversed for constants to prevent NPE